examples/optical_flow_methods_convergence.py

```diff
@@ -0,0 +1,363 @@
+
+    # NOTE: The motion field passed to the morph function can't have any NaNs.
+    # Otherwise, it can result in a segmentation fault.
+    morphed_field, mask = morph(input_precip.swapaxes(0, 1),
+                                ideal_motion.swapaxes(1, 2))
+
+    mask = np.array(mask, dtype=bool)
+
+    synthetic_observations = np.ma.MaskedArray(morphed_field, mask=mask)
+    synthetic_observations = synthetic_observations[np.newaxis, :]
+
+    for t in range(1, num_times):
+        morphed_field, mask = morph(synthetic_observations[t - 1],
+                                    ideal_motion.swapaxes(1, 2))
+        mask = np.array(mask, dtype=bool)
+
+        morphed_field = np.ma.MaskedArray(morphed_field[np.newaxis, :],
+                                          mask=mask[np.newaxis, :])
+
+        synthetic_observations = np.ma.concatenate([synthetic_observations, morphed_field],
+                                                   axis=0)
+
+    # Swap  back to (lat, lon)
+    synthetic_observations = synthetic_observations.swapaxes(1, 2)
+
+    synthetic_observations = np.ma.masked_invalid(synthetic_observations)
+
+    synthetic_observations.data[np.ma.getmaskarray(synthetic_observations)] = 0
+
+    return ideal_motion, synthetic_observations
+
+
+def plot_optflow_method_convergence(input_precip,
+                                    optflow_method_name,
+                                    motion_type):
+    """
+    Test the convergence to the actual solution of the optical flow method used.
+
+    Parameters
+    ----------
+
+    input_precip: numpy array (lat, lon)
+        Input precipitation field.
+
+    optflow_method_name: str
+        Optical flow method name
+
+    motion_type : str
+        The supported motion fields are:
+
+            - linear_x: (u=2, v=0)
+            - linear_y: (u=0, v=2)
+            - rotor: rotor field
+    """
+
+    if optflow_method_name != "DARTS":
+        num_times = 2
+    else:
+        num_times = 9
+
+    ideal_motion, precip_obs = create_observations(input_precip,
+                                                   motion_type,
+                                                   num_times=num_times)
+
+    oflow_method = motion.get_method(optflow_method_name)
+
+    computed_motion = oflow_method(precip_obs, verbose=False)
+
+    precip_data, _ = stp.utils.dB_transform(precip_obs.max(axis=0),
+                                            inverse=True)
+    precip_data.data[precip_data.mask] = 0
+
+    precip_mask = ((uniform_filter(precip_data, size=20) > 0.1)
+                   & ~precip_obs.mask.any(axis=0))
+
+    cmap = get_cmap('jet')
+    cmap.set_under('grey', alpha=0.25)
+    cmap.set_over('none')
+
+    # Compare retrieved motion field with the ideal one
+    plt.figure(figsize=(9, 4))
+    ax = plot_precip_field(input_precip, geodata=metadata,
+                           title="Reference motion",
+                           cartopy_subplot=(1, 2, 1), map='cartopy')
```

Comment:
Here `map='cartopy'` causes the doc build to fail, since cartopy is not an actual requirement, but only an optional dependency. I faced a similar problem with other examples. 
A simpler figure without any basemap will fix the issue, but we could also consider installing cartopy when building the documentation.

Reply:
Fix in 8b6eeef